external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/internal/OffsetManager.java

```diff
@@ -162,7 +164,7 @@ public long commit(OffsetAndMetadata committedOffset) {
         LOG.debug("Committed offsets [{}-{} = {}] for topic-partition [{}].",
                     preCommitCommittedOffsets + 1, this.committedOffset, numCommittedOffsets, tp);
         
-        return numCommittedOffsets;
+        return numCommittedOffsets - 1;// The committed offset should be the next message
```

Comment:
Please fix this in L144 instead, so the log messages above print the right number as well